### ISSUE TITLE
Bad debuggable args processing

### ISSUE THREAD

**Issue by AJIOB** [microsoft/vscode-cmake-tools#2898]
### Brief Issue Summary

I'm writing an app using GCC, CMake & VSCode. Sometimes I need to run my executable without attach debugger. And CMake Tools provides that with the "Launch selected target in the terminal window" button.

But it looks like CMake Tools doesn't correctly processing debug args:
* bad CMake/VSCode commands processing
* bad spaces & special symbols escaping

If I use "Launch debugger for the selected target", all works correctly and args are processed, as required:
```bash
--input1=/workspaces/PrintCommandLine/build
--input2=<paramName param=Value>
--input3=<paramName param=/workspaces/PrintCommandLine/build>
--input4=argument with space
```

If I use "Launch selected target in the terminal window", I have some errors in terminal:
```bash
$ /workspaces/PrintCommandLine/build/PrintCommandLine --input1=${command:cmake.launchTargetDirectory} --input2=<paramName param="Value"> --input3=<paramName param="${command:cmake.launchTargetDirectory}"> --input4=argument with space
bash: paramName: No such file or directory
```

Windows "Launch selected target" output:
```powershell
> "PrintCommandLine.exe" --input1=${command:cmake.launchTargetDirectory} --input2=<paramName param="Value"> --input3=<paramName param="${command:cmake.launchTargetDirectory}"> --input4=argument with space
--input1=
--input2=<paramName
param=Value>       
--input3=<paramName
param=>
--input4=argument  
with
space
```

Expected output (got from Windows debug):
```powershell
--input1=D:\Alex\Documents\SourceTree\PrintCommandLine\build\Debug
--input2=<paramName param="Value">
--input3=<paramName param="D:\Alex\Documents\SourceTree\PrintCommandLine\build\Debug">
--input4=argument with space
```

My `.vscode/settings.json` file:
```json
{
    "cmake.debugConfig": {
        "args": [
            "--input1=${command:cmake.launchTargetDirectory}",
            "--input2=<paramName param=\"Value\">",
            "--input3=<paramName param=\"${command:cmake.launchTargetDirectory}\">",
            "--input4=argument with space",
        ]
    },
}
```

No more CMake Tools configuration options are required for reproducing.

Project to test passing args: https://github.com/AJIOB/PrintCommandLine

My environment:
* OS: Debian 10 (Buster) - Linux-based, Windows 10 Pro 22H2
* Arch: amd64 (x86_64)
* Terminal: bash (linux), powershell (windows)
* VSCode: 1.74.1 (latest)
* CMake Tools Plug-in: v1.12.27 (latest)

### CMake Tools Diagnostics

```shell
// Linux output
{
  "os": "linux",
  "vscodeVersion": "1.74.1",
  "cmtVersion": "1.12.27",
  "configurations": [
    {
      "folder": "/workspaces/PrintCommandLine",
      "cmakeVersion": "3.21.5",
      "configured": true,
      "generator": "Ninja",
      "usesPresets": false,
      "compilers": {
        "C": "/usr/bin/gcc",
        "CXX": "/usr/bin/g++"
      }
    }
  ],
  "cpptoolsIntegration": {
    "isReady": false,
    "hasCodeModel": false,
    "activeBuildType": "",
    "buildTypesSeen": [],
    "requests": [],
    "responses": [],
    "partialMatches": [],
    "targetCount": 0,
    "executablesCount": 0,
    "librariesCount": 0,
    "targets": []
  },
  "settings": [
    {
      "communicationMode": "automatic",
      "useCMakePresets": "auto",
      "configureOnOpen": true
    }
  ]
}

// Windows output:
{
  "os": "win32",
  "vscodeVersion": "1.74.1",
  "cmtVersion": "1.12.27",
  "configurations": [
    {
      "folder": "d:\\Alex\\Documents\\SourceTree\\PrintCommandLine",
      "cmakeVersion": "3.16.5",
      "configured": true,
      "generator": "Visual Studio 16 2019",
      "usesPresets": false,
      "compilers": {}
    }
  ],
  "cpptoolsIntegration": {
    "isReady": false,
    "hasCodeModel": false,
    "activeBuildType": "",
    "buildTypesSeen": [],
    "requests": [],
    "responses": [],
    "partialMatches": [],
    "targetCount": 0,
    "executablesCount": 0,
    "librariesCount": 0,
    "targets": []
  },
  "settings": [
    {
      "communicationMode": "automatic",
      "useCMakePresets": "auto",
      "configureOnOpen": true
    }
  ]
}
```

### Debug Log

```shell
Not required for reproduction
```

### Additional Information

Not required for reproduction

**Comment by bobbrow** [comment]
In my case, the commands are being processed, but there is still the issue with spaces.  I have a fix.

**Comment by bobbrow** [comment]
Ah, yes, #2846 fixed the command processing, but hasn't been officially released yet.

**Comment by AJIOB** [comment]
@bobbrow 

Hello Bob.

What about the problems with possible unwanted stdin/stdout redirects with `<` or `>`?

That looks like bash & any other sh-compatible terminals problem.

Do you can reproduce it too?

**Comment by bobbrow** [comment]
With my fix, I believe it is working:
![image](https://user-images.githubusercontent.com/12818240/208753733-eb202f73-2ff1-4d4f-8895-11bbdb70e5ee.png)

**Comment by bobbrow** [comment]
The fix will be available in tomorrow's pre-release version (1.13.30).  You can switch to the pre-release via the VS Code Extensions sidebar panel.
